prov/efa/test/efa_unit_test_ep.c

```diff
@@ -12,19 +12,38 @@
  */
 void test_rxr_ep_host_id(struct efa_resource **state, bool file_exists, char *raw_id, uint64_t expect_id)
 {
+	int fd;
+	ssize_t written_len;
+	const size_t host_id_file_length = 10;
+	char *host_id_file;
 	struct efa_resource *resource = *state;
 	struct rxr_ep *rxr_ep;
-	size_t file_name_length = 10;
-	char host_id_file[file_name_length];
-	FILE *f_handle;
 
 	rxr_env.host_id_file = NULL;
 
 	if (file_exists) {
-		new_temp_file(host_id_file, file_name_length);
-		f_handle = fopen(host_id_file, "w");
-		fprintf(f_handle, raw_id);
-		fclose(f_handle);
+		host_id_file = malloc(host_id_file_length + 1);
+		assert_non_null(host_id_file);
+
+		for (int i = 0; i < host_id_file_length; ++i) {
+			host_id_file[i] = 'X';
+		}
+
+		host_id_file[host_id_file_length] = '\0';
+
+		fd = mkstemp(host_id_file);
```

Comment:
Could you elaborate? `mkstemp` could fail, do you suggest we do not need to check the failure? Also `close(fd)` makes the file inaccessible so it should only be called after the test if the file was actually created successfully.

Reply:
I was suggesting the highlighted section of code could be replaced with the first 3 lines of my snippet.  No need to catch an error for malloc (I guess just the one error).  Whatever you are doing with fd afterwards would remain the same, but if you can't mkstemp, then obviously the test will fail.

As far as closing the file, I think the logic you have is fine, just trying to point out you wouldn't need a free() as well as a close().

Reply:
I don't know how it could fail but `snprintf` could return a negative error code. Are you suggesting we ignore the failure and proceed to `mkstemp` anyways, assuming that the failure is not significant as long as `mkstemp` works?

Reply:
If you prefer to keep the loop or do a memset or whatever, that's fine.  Mostly I prefer to avoid the unnecessary alloc